### ISSUE TITLE
DynamoRIO bug preventing execution on AVX-512 capable CPUs

### ISSUE THREAD

**Issue by fmoessbauer** [siemens/drace#32]
Error observed on a machine with 80GB MEM
```
C:\opt\DynamoRIO-Windows-7.91.18137-0\bin64\drrun.exe -debug -- C:\Temp\x64-Release-TSAN\bin\test\mini-apps\concurrent-inc\gp-concurrent-inc.exe
<Starting application C:\Temp\x64-Release-TSAN\bin\test\mini-apps\concurrent-inc\gp-concurrent-inc.exe (3536)>
<Early threads found>
<Initial options = -no_dynamic_options -code_api -probe_api -stack_size 56K -max_elide_jmp 0 -max_elide_call 0 -no_inline_ignored_syscalls -native_exec_default_list '' -no_native_exec_managed_code -no_indcall2direct -pad_jmps_mark_no_trace >
<Application C:\Temp\x64-Release-TSAN\bin\test\mini-apps\concurrent-inc\gp-concurrent-inc.exe (3536) DynamoRIO usage error : encode error: rip-relative reference out of 32-bit reach>
<Usage error: encode error: rip-relative reference out of 32-bit reach (..\..\core\arch\x86\encode.c, line 3112)
version 7.91.18137, custom build
-no_dynamic_options -code_api -probe_api -stack_size 56K -max_elide_jmp 0 -max_elide_call 0 -no_inline_ignored_syscalls -native_exec_default_list '' -no_native_exec_managed_code -no_indcall2direct -pad_jmps_mark_no_trace >
```

**Comment by TomSie** [comment]
Hey,
we might be observing the same issue here with a machine of 128GB RAM. 
DynamoRIO-Windows-7.0.0-RC1 seems to work
DynamoRIO-Windows-7.1.0-1 seems to work sometimes
DynamoRIO-Windows-7.91.18207-0 (current nightly) fails

we are still investigating. Just wanted to let you know.

**Comment by fmoessbauer** [comment]
@TomSie : Thanks for letting me know. Can you confirm, that this is independent of AV / Crowdstrike?
Then I would open an issue in the DynamoRIO repository. Most times, they respond quite fast.

BTW: at least I run on DR on Windows. Does the issue also appear on Linux? Would be best to put as much information in the bug report as possible.

**Comment by TomSie** [comment]
I can confirm, that the issue is independent from AV / Crowdstrike and therelike. Our test system is a fresh Windows Server 2016 without any additional software.

We are investigating if we can narrow the issue down to a certain amount of RAM.

**Comment by jones718** [comment]
I have tested the version DynamoRIO-Windows-7.91.18207-0 in a virtual machine with 122 GB memory and it still worked. I will further investigate this problem by trying out older releases.

**Comment by jones718** [comment]
I tried older releases and found out that cronbuild-7.91.18109 is the first version that's not working at all on our machine. It's causing a total crash of the program (tested with notepad.exe). Furthermore the release cronbuild-7.0.17598 is the first one only working partially and causing the following error message: "Application path\notepad.exe. Failed to take over all threads after multiple attempts". 
I will go on by trying several commits of these releases.

**Comment by fmoessbauer** [comment]
The actual bug is not memory related but due to a bug in DynamoRIO when running on a AVX-512 capable CPU. A quick workaround is to disable the AVX-512 optimization by setting this flag to false:
https://github.com/DynamoRIO/dynamorio/blob/cf1ec32e9b89c1d8a28e0f355b8338d4c05c0f14/core/arch/x86/proc.c#L437

I opened a issue in the DynamoRIO repo to continue the investigation there: xref https://github.com/DynamoRIO/dynamorio/issues/3949

**Comment by fmoessbauer** [comment]
Bug is fixed in 7.91.18219
